core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -509,16 +507,13 @@ private static void checkUri(AlluxioURI uri) {
       /* Even if we choose to log the warning, check if the Configuration host matches what the
        * user passes. If not, throw an exception letting the user know they don't match.
        */
-      String host = Configuration.get(PropertyKey.MASTER_HOSTNAME);
-      int port = Configuration.getInt(PropertyKey.MASTER_RPC_PORT);
-      Authority trueAuthority = new SingleMasterAuthority(host, port);
-      if (!trueAuthority.equals(uri.getAuthority())) {
-        throw new IllegalArgumentException(String.format(
-            "The URI authority %s does not match the configured "
-            + "value of %s.", uri.getAuthority(), trueAuthority));
+      Authority configured = MasterInquireClient.Factory.create().getConnectDetails().toAuthority();
+      if (!configured.equals(uri.getAuthority())) {
```

Comment:
Would it be possible that the user passed in something like `127.0.0.1` but we use `localhost` in the connection details or vice versa?

In this circumstances, even if they are different in appearance, they are the same internally.

Reply:
You bring up a good point. If multiple hostnames map to the same IP, or the user mixes IP with hostnames, this check could fail when it doesn't need to. I think we might eventually implement a good mechanism for comparing addresses that is more forgiving than strict `String` equality. But that's a much bigger issue that would apply across the codebase to all the many places we compare address strings. We also need to be careful about the potential performance impact of doing a lot of DNS lookups, though this can probably be solved with caching.